### PR TITLE
Import neon-mentat

### PR DESCRIPTION
**Pull request by bgrins**
What this does is set up a dependency for neon-mentat, which allows us to run mentat code in the browser server.  It also has travis run with rust installed and reorganizes the config a bit to accommodate that builds for this branch are at (https://www.travis-ci.org/victorporof/tofino/branches).

It exposes two main functions: `query` and `transact`.  Everything is done in memory for now but eventually you'll be able to pass a string to the constructor for saving to a db.

This doesn't hook up any actual logic beyond a test yet, because building correctly on windows requires some extra work and I want to be careful to not break people's workflow if possible.  What happens if you are on windows without node-gyp or don't have rust installed (on any platform) is the mentat object that gets used is a stubbed out function - so it doesn't break but it also doesn't do anything: https://github.com/bgrins/neon-mentat/blob/master/lib/connection.js.

### CODE REVIEW

**Comment by bgrins** [comment]
@bwinton has kindly volunteered to check and make sure this doesn't break the build on his machine.

The longer term plan to ultimately support and get this running on windows is to install node-gyp as outlined in https://github.com/neon-bindings/neon#requirements

File: src/browser-server/mentat.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (victorporof) [inline src/browser-server/mentat.js:12]: License header here.

File: src/browser-server/test/test-mentat.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import expect from 'expect';
```
Comment (victorporof) [inline src/browser-server/test/test-mentat.js:4]: Test license header here please.

**Review by victorporof — APPROVED** [review]
Brian is killing it.

**Comment by victorporof** [comment]
Land at will.

**Comment by bwinton** [comment]
Running `npm run test` on Windows gives:
``` 7 passing (436ms)
  1 failing

  1) mentat should query:

      Error: Expected 0 to equal 1
      + expected - actual

      -0
      +1

      at assert (node_modules\expect\lib\assert.js:29:9)
      at Expectation.toEqual (node_modules\expect\lib\Expectation.js:81:30)
      at Context.<anonymous> (C:/Users/bwinton/Programming/tofino/src/browser-server/test/test-mentat.js:25:34)
```
so, uh, it would be nice if we could get that working before landing.  😉

I also get a bunch of:
```error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-pc-windows-msvc` target may not be installed
…
error: Could not compile `libc`.
```
errors running `npm run package` which we should probably fix somehow.  (Maybe by adding rust to our install instructions?  Maybe by making it more optional somehow?)

**Review by bwinton — CHANGES_REQUESTED** [review]
(Changes as described in my previous comment.)

**Comment by victorporof** [comment]
@bwinton https://github.com/victorporof/tofino/issues/37 for the win?

**Comment by bwinton** [comment]
That would definitely help, if someone wanted to set it up…  🙂

**Comment by bgrins** [comment]
I'll see about setting up appveyor on this branch

**Comment by bgrins** [comment]
OK @bwinton I think the latest changes on this branch should build and pass `npm test` on windows.  Can you let me know if it works for you?

I also have windows CI running at https://ci.appveyor.com/project/bgrins/tofino, which should be actually building and running the native module.  It also does a package so it should fail if packaging breaks - next step would be to deploy the artifacts somewhere.

**Comment by bwinton** [comment]
Whoa, conflicts…  Did you want to push a rebased version and I'll test that?  😉

**Comment by bgrins** [comment]
> Whoa, conflicts… Did you want to push a rebased version and I'll test that?

Done

**Review by bwinton — APPROVED** [review]

**Comment by bwinton** [comment]
Looks _way_ better.  Thank you!

**Pull request merged**